From 48d7f5314be4903c6ffc2fb4be542d61b7241b8b Mon Sep 17 00:00:00 2001
From: Aaron Levy <aaron.levy@coreos.com>
Date: Thu, 19 Nov 2015 19:06:39 -0800
Subject: [PATCH 2/4] explicitly check "Ready" condition in validate-cluster

---
 cluster/validate-cluster.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cluster/validate-cluster.sh b/cluster/validate-cluster.sh
index 7b26295..9756439 100755
--- a/cluster/validate-cluster.sh
+++ b/cluster/validate-cluster.sh
@@ -40,7 +40,7 @@ while true; do
   # Suppress errors from kubectl output because during cluster bootstrapping
   # for clusters where the master node is registered, the apiserver will become
   # available and then get restarted as the kubelet configures the docker bridge.
-  nodes_status=$("${KUBE_ROOT}/cluster/kubectl.sh" get nodes -o template --template='{{range .items}}{{with index .status.conditions 0}}{{.type}}:{{.status}},{{end}}{{end}}' --api-version=v1) || true
+  nodes_status=$("${KUBE_ROOT}/cluster/kubectl.sh" get nodes -o template --template='{{range .items}}{{range .status.conditions}}{{if eq .type "Ready"}}{{.type}}:{{.status}},{{end}}{{end}}{{end}}' --api-version=v1) || true
   found=$(echo "${nodes_status}" | tr "," "\n" | grep -c 'Ready:') || true
   ready=$(echo "${nodes_status}" | tr "," "\n" | grep -c 'Ready:True') || true
 
-- 
2.3.8 (Apple Git-58)

